rust/cardano-chain-follower/src/mithril_turbo_downloader.rs

```diff
@@ -147,22 +145,33 @@ impl Inner {
                         buf.len()
                     );
                 }
-                // Got the full file and its the expected size.  Is it different?
-                if memcmp(prev_mmap.as_slice(), buf.as_slice()) == cmp::Ordering::Equal {
-                    // Same so lets Hardlink it, and throw away the temp buffer.
 
-                    // Make sure our big mmap get dropped.
-                    drop(prev_mmap);
-
-                    // File is the same, so dedup it.
-                    if self.cfg.dedup_tmp(&abs_file, &latest_snapshot).is_ok() {
-                        self.dedup_size.fetch_add(entry_size, Ordering::SeqCst);
-                        changed_file!(self, rel_file, abs_file, entry_size);
-                        drop(buf);
-                        continue;
+                let mut prev_mmap_option = Some(prev_mmap);
+                // Take the value inside the option out
+                if let Some(prev_mmap) = prev_mmap_option.take() {
+                    if memcmp(prev_mmap.file_as_slice(), buf.as_slice()) == cmp::Ordering::Equal {
+                        // Same so let's hardlink it, and throw away the temp buffer.
+                        // Make sure our big mmap get dropped.
+                        drop(prev_mmap);
+                        stats::set_mmap_drop(self.cfg.chain);
+
+                        // File is the same, so dedup it.
+                        if self.cfg.dedup_tmp(&abs_file, &latest_snapshot).is_ok() {
+                            self.dedup_size.fetch_add(entry_size, Ordering::SeqCst);
+                            changed_file!(self, rel_file, abs_file, entry_size);
+                            drop(buf);
+                            continue;
+                        }
                     }
                 }
 
+                // If the `prev_mmap` is not yet drop, drop it now.
+                // Need to do this way because drop is moved into the if block.
+                if let Some(prev_mmap) = prev_mmap_option {
```

Comment:
Because of thje above `take`, prev_mmap_option must always be None here.  Not understanding whats attempted here?